### PR TITLE
feat(TDP-3607): check storage quota

### PR DESCRIPTION
**Pull request by ncomont**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-3607

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/555/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/555/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/557/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/557/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/558/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/558/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/641/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/641/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/646/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/646/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/648/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/648/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/649/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/649/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/653/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/653/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/655/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/655/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/659/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/659/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/660/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/660/)

File: dataprep-api/src/main/java/org/talend/dataprep/api/service/command/dataset/CreateOrUpdateDataSet.java [hunk 1]
```diff
@@ -45,22 +44,28 @@
...
             } catch (URISyntaxException e) {
                 throw new TDPException(CommonErrorCodes.UNEXPECTED_EXCEPTION, e);
             }
         });
-        onError(e -> new TDPException(APIErrorCodes.UNABLE_TO_CREATE_OR_UPDATE_DATASET, e));
+        onError(e -> {
+            if (e instanceof TDPException) {
+                return passthrough().apply(e);
+            }
+            return new TDPException(APIErrorCodes.UNABLE_TO_CREATE_DATASET, e);
```
[the remark this answers is not in the document]
Comment (vjaouen) [reply dataprep-api/src/main/java/org/talend/dataprep/api/service/command/dataset/CreateOrUpdateDataSet.java]: :+1:

File: dataprep-dataset/src/test/java/org/talend/dataprep/dataset/service/DataSetServiceTest.java [hunk 2]
```diff
@@ -428,6 +470,58 @@ public void cannotCreateWhenNameIsAlreadyUsed() throws Exception {
...
+        // when
+        final Response response = given() //
+                .body(IOUtils.toString(this.getClass().getResourceAsStream(T_SHIRT_100_CSV), UTF_8)) //
+                .queryParam("Content-Type", "text/csv") //
+                .queryParam("name", "kamoulox") //
+                .queryParam("size", -1L) // negative size value
+                .post("/datasets");
+
+        // then
+        assertEquals(413, response.getStatusCode());
```
[the remark this answers is not in the document]
Comment (vjaouen) [reply dataprep-dataset/src/test/java/org/talend/dataprep/dataset/service/DataSetServiceTest.java]: :+1:

File: dataprep-dataset/src/test/java/org/talend/dataprep/dataset/store/metadata/file/FileSystemDataSetMetadataRepositoryTest.java [hunk 3]
```diff
@@ -114,7 +115,8 @@ public void shouldReturnNullWhenGetEncounterAnError() throws Exception {
             // then
             assertNull(actual);
         } finally {
-            metadataFile.delete();
+            boolean isDeleted = metadataFile.delete();
```
[the remark this answers is not in the document]
Comment (vjaouen) [reply dataprep-dataset/src/test/java/org/talend/dataprep/dataset/store/metadata/file/FileSystemDataSetMetadataRepositoryTest.java]: :+1:

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/661/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/661/)

**Pull request merged**